server/monitor.go

```diff
@@ -0,0 +1,504 @@
+	tsecs := d / time.Second
+	tmins := tsecs / 60
+	thrs := tmins / 60
+	tdays := thrs / 24
+	tyrs := tdays / 365
+
+	if tyrs > 0 {
+		return fmt.Sprintf("%dy%dd%dh%dm%ds", tyrs, tdays%365, thrs%24, tmins%60, tsecs%60)
+	}
+	if tdays > 0 {
+		return fmt.Sprintf("%dd%dh%dm%ds", tdays, thrs%24, tmins%60, tsecs%60)
+	}
+	if thrs > 0 {
+		return fmt.Sprintf("%dh%dm%ds", thrs, tmins%60, tsecs%60)
+	}
+	if tmins > 0 {
+		return fmt.Sprintf("%dm%ds", tmins, tsecs%60)
+	}
+	return fmt.Sprintf("%ds", tsecs)
+}
+
+func (s *StanServer) handleStorez(w http.ResponseWriter, r *http.Request) {
+	count, bytes, _ := s.store.MsgsState(stores.AllChannels)
+	storez := &Storez{
+		ClusterID:  s.info.ClusterID,
+		ServerID:   s.serverID,
+		Now:        time.Now(),
+		Type:       s.store.Name(),
+		Limits:     *s.storeLimits,
+		TotalMsgs:  count,
+		TotalBytes: bytes,
+	}
+	sendResponse(w, r, storez)
+}
+
+type byClientID []*Clientz
+
+func (c byClientID) Len() int           { return len(c) }
+func (c byClientID) Swap(i, j int)      { c[i], c[j] = c[j], c[i] }
+func (c byClientID) Less(i, j int) bool { return c[i].ID < c[j].ID }
+
+func (s *StanServer) handleClientsz(w http.ResponseWriter, r *http.Request) {
+	singleClient := r.URL.Query().Get("client")
+	subsOption, _ := strconv.Atoi(r.URL.Query().Get("subs"))
+	if singleClient != "" {
+		var clientz *Clientz
+		client := s.store.GetClient(singleClient)
+		if client != nil {
+			clientz = getMonitorClient(client, subsOption)
+		}
+		if clientz == nil {
+			w.WriteHeader(http.StatusNotFound)
+			w.Write([]byte(fmt.Sprintf("Client %s not found", singleClient)))
+			return
+		}
+		sendResponse(w, r, clientz)
+	} else {
+		offset, _ := strconv.Atoi(r.URL.Query().Get("offset"))
+		if offset < 0 {
+			offset = 0
+		}
+		limit, _ := strconv.Atoi(r.URL.Query().Get("limit"))
+		if limit <= 0 {
+			limit = defaultMonitorListLimit
+		}
+		clients := s.store.GetClients()
+		totalClients := len(clients)
+		carr := make([]*Clientz, 0, totalClients)
+		for _, c := range clients {
+			cz := &Clientz{ID: c.ID}
+			carr = append(carr, cz)
+		}
+		sort.Sort(byClientID(carr))
+
+		minoff := offset
+		if minoff > totalClients {
+			minoff = totalClients
+		}
+		maxoff := offset + limit
+		if maxoff > totalClients {
+			maxoff = totalClients
+		}
+		carr = carr[minoff:maxoff]
+
+		for _, c := range carr {
+			cli := clients[c.ID]
+			c.HBInbox = cli.HbInbox
+			if subsOption == 1 {
+				srvCli := cli.UserData.(*client)
+				c.Subscriptions = getMonitorClientSubs(srvCli, true)
+			}
+		}
+		clientsz := &Clientsz{
+			ClusterID: s.info.ClusterID,
+			ServerID:  s.serverID,
+			Now:       time.Now(),
+			Offset:    offset,
+			Limit:     limit,
+			Total:     totalClients,
+			Count:     len(carr),
+			Clients:   carr,
+		}
+		sendResponse(w, r, clientsz)
+	}
+}
+
+func getMonitorClient(c *stores.Client, subsOption int) *Clientz {
+	cli := c.UserData.(*client)
+	cli.RLock()
+	if cli.unregistered {
+		cli.RUnlock()
```

Comment:
Seems like a single `defer cli.RUnlock()` would be appropriate in this function since it's not really a hotpath.